File: medium/numbers/number.php
<!DOCTYPE html>
<html>
<head>
	<title>Numbers</title>
</head>
<body>
	<form  method="post">
		<label for="text">Text : </label>
		<input type="text"  name="text">

		<input type="submit" value="Action" >
	</form>
	<br>
	<?php

		function number_conversion($text){

			if ($text == NULL ){
				return("");
			}

			$dict_numbers = array('0'=>'zero','1'=>'one',
			'2'=>'two','3'=>'three',
			'4'=>'four','5'=>'five','6'=>'six',
			'7'=>'seven','8'=>'eight','9'=>'nine');

			$result = "";

			for ($i=0; $i < strlen($text) ; $i++) { 


				if(array_key_exists($text[$i], $dict_numbers)) {
					
					$result = $result.$dict_numbers[$text[$i]];
				}
				else{

					$result = $result.$text[$i];
				}
			}

			
			echo $result;	
			
		}

	    $text = $_POST["text"];

	    number_conversion($text);
        
	?>

</body>
</html>
		

File: medium/numbers/number.c
#include <stdio.h>
#include <string.h>

void main(){
    char text[100];
    fgets(text,100,stdin);

    char numbers[10][6] = {"zero","one","two","three","four","five","six","seven","eight","nine"};


    for(int i = 0;i < strlen(text) ;i++){
        if(text[i] >= 48 && text[i] <= 57 ){
            if (text[i] == 49 && text[i+1] == 48){
                printf("%s", "ten");
                i++;
            }
            else  printf("%s", numbers[(int)text[i] - 48 ]);

        }
        else printf("%c",text[i]);
    }
}
